Add documentation for xk6-disruptor (#891)
grafana/k6-docs: src/data/markdown/docs/40 xk6-disruptor/02 API/02 PodDisruptor.md

---
title: 'PodDisruptor'
excerpt: 'xk6-disruptor: PodDisruptor class'
---

> The `PodDisruptor` class allows the injection of different types of faults in the pods that match a selection criteria.

Reviewer: ```suggestion
The `PodDisruptor` class can inject different types of faults into the pods that match a selection criteria.
```

To construct a `PodDisruptor`, use the [PodDisruptor() constructor](/javascript-api/xk6-disruptor/api/poddisruptor/constructor).

## Methods

| Method | Description |
| ------ | ----------- |
| [PodDisruptor.injectHTTPFaults()](/javascript-api/xk6-disruptor/api/poddisruptor/injecthttpfaults) |  Inject [HTTTP faults](/javascript-api/xk6-disruptor/api/faults/http) in the target Pods|
| PodDisruptor.targets() | Returns the list of target Pods of the PodDisruptor |
 

## Example

> This example shows how to create a selector that matches all pods in the `default` namespace with the `run=nginx` label and inject a delay of 100ms and a 10% of requests returning a http response code 500.

Reviewer: ```suggestion
This example:
- Creates a selector that matches all pods in the `default` namespace with the `run=nginx` label
- Injects a delay of 100ms and makes 10 percent of requests return an http response code `500`.
```
